plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -950,181 +977,149 @@ public function get_field_from_customer( $key, $customer, $group = '' ) {
 	 *
 	 * @param string   $field The field key.
 	 * @param WC_Order $order The order to get the field value for.
-	 * @param string   $group The group to get the field value for (shipping|billing|'') in which '' refers to the additional group.
+	 * @param string   $group The group to get the field value for (shipping|billing|additional) and defaults to additional.
 	 *
 	 * @return mixed The field value.
 	 */
-	public function get_field_from_order( $field, $order, $group = '' ) {
+	public function get_field_from_order( string $field, WC_Order $order, string $group = 'additional' ) {
 		return $this->get_field_from_object( $field, $order, $group );
 	}
 
 	/**
 	 * Returns a field value for a given object.
 	 *
 	 * @param string               $key The field key.
-	 * @param WC_Customer|WC_Order $object The customer to get the field value for.
-	 * @param string               $group The group to get the field value for (shipping|billing|'') in which '' refers to the additional group.
+	 * @param WC_Customer|WC_Order $wc_object The customer to get the field value for.
+	 * @param string               $group The group to get the field value for (shipping|billing|additional).
 	 *
 	 * @return mixed The field value.
 	 */
-	private function get_field_from_object( $key, $object, $group = '' ) {
-		$meta_key = '';
-		if ( 0 === strpos( $key, '/billing/' ) || 'billing' === $group ) {
-			$meta_key = self::BILLING_FIELDS_KEY;
-			$key      = str_replace( '/billing/', '', $key );
-		} elseif ( 0 === strpos( $key, '/shipping/' ) || 'shipping' === $group ) {
-			$meta_key = self::SHIPPING_FIELDS_KEY;
-			$key      = str_replace( '/shipping/', '', $key );
-		} else {
-			$meta_key = self::ADDITIONAL_FIELDS_KEY;
-		}
-
-		$meta_data = $object->get_meta( $meta_key, true );
+	private function get_field_from_object( string $key, WC_Data $wc_object, string $group = 'additional' ) {
+		$meta_key = self::get_group_key( $group ) . $key;
 
-		if ( ! is_array( $meta_data ) ) {
-			return '';
-		}
+		$meta_data = $wc_object->get_meta( $meta_key, true );
 
-		if ( ! isset( $meta_data[ $key ] ) ) {
+		if ( '' === $meta_data || null === $meta_data ) {
```

Comment:
```suggestion
		if ( null === $meta_data ) {
```

We're already returing that value regardless